.github/workflows/ci.yml

```diff
@@ -12,23 +12,47 @@ on:
 jobs:
   build:
 
-    runs-on: macos-latest
-
+    runs-on: #ubuntu-latest
+      - self-hosted
+      - eclipse
+    
     steps:
     - uses: actions/checkout@v2
+
+    - name: Clone IDF Release From Github
+      uses: actions/checkout@v2
+      with:
+        repository: espressif/esp-idf
+        path: dependencies/idf-tools
+        submodules: 'true'
+        ref: v4.4
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+          python-version: '3.9'
+
+    - name: Set up Maven
+      uses: stCarolas/setup-maven@v4.2
+      with:
+        maven-version: 3.8.4
    
     - name: Set up JDK 11
       uses: actions/setup-java@v2
       with:
         java-version: '11'
         distribution: 'adopt'
-   
+
     - name: Build with Maven
-      run: mvn clean verify -Djarsigner.skip=true
+      run: mvn clean verify -Djarsigner.skip=true -Dtycho.showEclipseLog=true
+
+    - name: Publish Test Reports
+      if: ${{ always() }}
+      uses: scacap/action-surefire-report@v1
   
     - name: Create macosx dmg
-      run: $PWD/releng/ide-dmg-builder/ide-dmg-builder.sh
-
```

Comment:
Can we change to system agnostic? Folder like `opt/actions-runner/_work/`

Reply:
tried with that but our runner was not catering any agnostic paths and I needed to point via absolute path